llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -2378,11 +2378,13 @@ let Predicates = [HasSVEorSME] in {
     def LDR_ZZXI   : Pseudo<(outs   ZZ_b_strided_and_contiguous:$Zd), (ins GPR64sp:$sp, simm4s1:$offset),[]>, Sched<[]>;
     def LDR_ZZZXI  : Pseudo<(outs  ZZZ_b:$Zd), (ins GPR64sp:$sp, simm4s1:$offset),[]>, Sched<[]>;
     def LDR_ZZZZXI : Pseudo<(outs ZZZZ_b_strided_and_contiguous:$Zd), (ins GPR64sp:$sp, simm4s1:$offset),[]>, Sched<[]>;
+    def LDR_PPXI   : Pseudo<(outs PPR2:$pp), (ins GPR64sp:$sp, simm4s1:$offset),[]>, Sched<[]>;
```

Comment:
`simm4s1` seems weird given the underlying instructions use `simm9` but then we don't target the pseudos via ISEL so I guess it doesn't really matter.  It's also consistent with the existing pseudos, which also look wrong.

Reply:
These offsets are checked when replacing frame index with actual addressing mode, so we know we are in range. I would suggest changing all these to `simm9` in a separate patch, if only to not make someone wonder why did we use such strange constraints.